mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -308,7 +308,7 @@ def _get_temporary_model_version_write_credentials(self, name, version) -> Tempo
         ).credentials
 
     def _get_workspace_id(self, run_id):
-        if run_id is None:
+        if run_id is None or not is_databricks_uri(self.tracking_uri):
```

Comment:
This method makes an API request to a Databricks-hosted tracking server and reads the `X-Databricks-Org-Id` header to determine the tracking server's workspace ID. Previously, we didn't actually check whether the current tracking server is a Databricks-hosted one, leading to errors here when the tracking URI was set to something other than `databricks[://profile]`. Here we explicitly check